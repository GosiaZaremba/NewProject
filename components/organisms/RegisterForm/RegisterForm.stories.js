/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { RegisterForm } from './RegisterForm';

storiesOf('organisms/RegisterForm', module)
    .addDecorator((story) => story())
    .add('default', () => <RegisterForm onSubmitForm={() => {}} />);
