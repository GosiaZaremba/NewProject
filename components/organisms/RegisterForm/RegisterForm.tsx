import React, { useRef } from 'react';
import { View, Switch } from 'react-native';
import { Colors } from '../../../constants/colors';
import {
    CustomButton,
    CustomInputReference,
    Label,
    CustomInput,
} from '../../atoms';
import styles from './RegisterForm.styles';

export type Props = {
    onSubmitForm: (email: string, password: string) => void;
};

export const RegisterForm: React.FC<Props> = ({ onSubmitForm }) => {
    const emailInputReference = useRef<CustomInputReference>(null);
    const passwordInputReference = useRef<CustomInputReference>(null);

    const onSubmit = () => {
        const emailValue = emailInputReference.current?.getValue() || '';
        const passwordValue = passwordInputReference.current?.getValue() || '';
        // console.log('email', emailValue, 'password', passwordValue);
        onSubmitForm(emailValue, passwordValue);
    };

    return (
        <View style={styles.outerContainer}>
            <View style={styles.inputsContainer}>
                <View style={styles.inputContainer}>
                    <Label text={'Email'} />
                    <CustomInput
                        placeholder={'email'}
                        secureTextEntry={false}
                        ref={emailInputReference}
                    />
                </View>
                <View style={styles.inputContainer}>
                    <Label text={'Password'} />
                    <CustomInput
                        placeholder={'password'}
                        secureTextEntry={true}
                        ref={passwordInputReference}
                    />
                </View>
            </View>
            <View style={styles.buttonContainer}>
                <CustomButton
                    title={'SIGN UP'}
                    onPress={onSubmit}
                    color={Colors.pink}
                />
            </View>
        </View>
    );
};
