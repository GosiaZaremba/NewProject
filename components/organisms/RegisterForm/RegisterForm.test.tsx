import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import { RegisterForm } from './RegisterForm';

const mockOnPress = jest.fn();

describe('Organism RegisterForm', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(<RegisterForm onSubmitForm={mockOnPress} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure submit button works correctly', () => {
        const { getByTestId } = render(
            <RegisterForm onSubmitForm={mockOnPress} />
        );

        const submitButton = getByTestId('organisms-registerForm-button');
        fireEvent.press(submitButton);
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        expect(submitButton).toHaveTextContent('SIGN UP');
    });

    test('ensure email input works correctly', () => {
        const { getByTestId } = render(
            <RegisterForm onSubmitForm={mockOnPress} />
        );

        const inputEmail = getByTestId('organisms-registerForm-email-input');
        fireEvent.changeText(inputEmail, 'mail@wp.pl');
        expect(inputEmail.props.value).toBe('mail@wp.pl');
    });

    test('ensure password input works correctly', () => {
        const { getByTestId } = render(
            <RegisterForm onSubmitForm={mockOnPress} />
        );

        const inputPassword = getByTestId(
            'organisms-registerForm-password-input'
        );
        fireEvent.changeText(inputPassword, 'haslo');
        expect(inputPassword.props.value).toBe('haslo');
    });

    test('ensure onSubmitForm works correctly', () => {
        const { getByTestId } = render(
            <RegisterForm onSubmitForm={mockOnPress} />
        );

        const inputPassword = getByTestId(
            'organisms-registerForm-password-input'
        );
        const inputEmail = getByTestId('organisms-registerForm-email-input');
        const submitButton = getByTestId('organisms-registerForm-button');
        fireEvent.changeText(inputPassword, 'haslo');
        fireEvent.changeText(inputEmail, 'mail@wp.pl');
        fireEvent.press(submitButton);
        expect(mockOnPress).toHaveBeenCalledWith('mail@wp.pl', 'haslo');
    });
});
