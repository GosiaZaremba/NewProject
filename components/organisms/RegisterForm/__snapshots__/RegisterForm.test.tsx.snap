// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism RegisterForm Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "black",
              "marginBottom": 2,
            }
          }
          testID="organism-registerForm-email-input-label"
        >
          Email
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="email"
          secureTextEntry={false}
          style={
            Object {
              "borderColor": "#908D8E",
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID="organism-registerForm-email-input-input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "black",
              "marginBottom": 2,
            }
          }
          testID="organism-registerForm-password-input-label"
        >
          Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="password"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#908D8E",
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID="organism-registerForm-password-input-input"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 15,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="SIGN UP"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="organism-registerForm-submit-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#fc296f",
              },
            ]
          }
        >
          SIGN UP
        </Text>
      </View>
    </View>
  </View>
</View>
`;
